ts/client/storageEngine/interfaces.ts

```diff
@@ -10,6 +10,8 @@ export interface Entry<Item> {
     item: Item;
 }
 
+// TODO Update this interface and implementations to
+// use BigNumber for itemID.
```

Comment:
Avoided this refactor as this PR is already large and didn't want to bork @kautukkundan 's work :) `BigNumberish` should be fine as well as long as the storage engine and tree both store in a large value format under the covers.

Reply:
oh makes sense, can discuss this